calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java

```diff
@@ -2713,6 +2723,20 @@ public CalendarEvent addEvent(TimeRange range, String displayName, String descri
 			return edit;
 
 		} // addEvent
+
+		/**
+		 * @return only the groups allowed and selected
+		 */
+		private Collection<Group> groupAllowedSelect (Collection<Group> group, Collection<Group> groupAllowed) {
+			return groupAllowed.stream().filter(it -> (group.contains(it))).collect(Collectors.toList());
+		}
+
+		/**
+		 * @return groups not allowed and selected
+		 */
+		private Collection<Group> groupNotAllowedSelect (Collection<Group> group, Collection<Group> groupAllowed) {
```

Comment:
Very minor and optional, I kinda don't like the function name, better if we provide something more clear like:

filterNotAllowedAndSelectedGroups